### PR TITLE
Fix for filtering images on image selector

### PR DESCRIPTION
**Pull request by hassan-mustafa-baig**
### Proposed Changes
* Image filtering has been fixed for image selector

### Checklist
- [ ] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
** any additional useful context or info **

### Screenshots
Original             |  Updated
:-------------------------:|:-------------------------:

image

image

image

### CODE REVIEW

**Review by jcastro-dotcms — CHANGES_REQUESTED** [review]

File: dotCMS/src/main/java/com/dotmarketing/portlets/browser/ajax/BrowserAjax.java [hunk 1]
```diff
@@ -503,6 +503,7 @@ public Map<String, Object> getFolderContentWithDotAssets(final String folderId,
 						   .maxResults(maxResults)
 						   .withFilter(filter)
 						   .showMimeTypes(mimeTypes)
```
[the remark this answers is not in the document]
Comment (hassan-mustafa-baig) [reply dotCMS/src/main/java/com/dotmarketing/portlets/browser/ajax/BrowserAjax.java:505]: we need to check image specifically therefore specific method is created for image

[the remark this answers is not in the document]
Comment (jcastro-dotcms) [reply dotCMS/src/main/java/com/dotmarketing/portlets/browser/ajax/BrowserAjax.java:505]: But this MIME Type should come from the JS or JSP instead, shouldn't it?

**Review by jcastro-dotcms — CHANGES_REQUESTED** [review]

File: dotCMS/src/main/java/com/dotcms/browser/BrowserQuery.java [hunk 2]
```diff
@@ -300,6 +299,13 @@ public Builder showDotAssets(@Nonnull boolean dotAssets) {
             return this;
         }
 
+        public Builder showImages(@Nonnull List<String> mimeTypes) {
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/browser/BrowserQuery.java:302]: show methods used to receive a boolean, such as 
```

        public Builder showImages(boolean showImages) {
        ```

The responsability of the calculation of the boolean should be outside of the showImages method

File: dotCMS/src/main/java/com/dotcms/browser/BrowserAPIImpl.java [hunk 3]
```diff
@@ -275,10 +275,11 @@ private Tuple2<String, List<Object>> selectQuery(final BrowserQuery browserQuery
             sqlQuery.append(" and id.parent_path=? ");
             parameters.add(browserQuery.folder.getPath());
         }
-        if (UtilMethods.isSet(browserQuery.filter)) {
-            final String[] spliter = browserQuery.filter.split(" ");
+        if (UtilMethods.isSet(browserQuery.luceneQuery)) {
+            final String filterText = browserQuery.luceneQuery.toLowerCase();
+            final String[] spliter = filterText.split(" ");
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/browser/BrowserAPIImpl.java:280]: shouldn't sanitize it, if it is not already sanitized

**Review by jcastro-dotcms — APPROVED** [review]

**Review by jdotcms — APPROVED** [review]

**Pull request merged**